QuoridorEngine/src/core/QuoridorGame.cs

```diff
@@ -79,13 +98,31 @@ public void ExecuteMove(Move move)
         }
 
         /// <summary>
-        /// Undoes a given move returning the state to its previous configuration. Assumes the move to be
-        /// undone was legal at the moment it was executed.
+        /// Undoes last given move returning the state to its previous configuration. 
+        /// Assumes the move to be undone was legal at the moment it was executed.
         /// </summary>
-        /// <param name="move">The move to be undone</param>
         public void UndoMove(Move move)
         {
-            throw new NotImplementedException();
+            QuoridorMove lastMove = (QuoridorMove)(move);
+            if (lastMove.Type == MoveType.PlayerMovement)
+            {
+                // Handle player movement undo
+            }
+            else if(lastMove.Type == MoveType.WallPlacement)
```

Comment:
Add some assertions to make sure values are inside array bounds. If for some reason the program gets invalid data we need to catch and fix the bug before anyone else uses it